### DISCUSSION TITLE
.NET Core 3.1 is there a removal date?

### DISCUSSION THREAD

**Discussion by markatky46** [actions/virtual-environments#7559] [Installed software]
Hi, We are being asked why an end of life version of .NET Core SDK  and tools (3.1) are still being deployed in our image. It is causing critical vulnerabilities on our scanning tools.

Is there a date to remove the .NET 3.1 from the .NET Core Tools?

**Comment by markatky46** [comment]
.NET Core | 2 latest LTS versions and 1 latest version. For each feature version only latest patch is installed

**Reply by ilia-shipitsin** [reply]
can you please provide more details on how do you invoke scanning tools ?

**Reply by markatky46** [reply]
Qualys scans a deployed image daily. I assume it works in the same way as Nessus and scans installed components.

**Comment by shamil-mubarakshin** [comment]
Hello @markatky46. We cannot remove .Net Core 3.1 yet as it will break some users who still use it. Its removal will be announced separately

**Reply by markatky46** [reply]
But it is totally end of life, out of support and has huge vulnerabilities. It also goes against your published strategy.

For the users who need it they should/could install via pipeline still. There is no reliance on this version for the agent anymore either. V3 of the agent now uses .Net 6

I can't be the only end user consuming this that is getting hammered off our security team for having end of life, unsupported and highly vulnerable build tools installed

**Reply by markatky46** [reply]
<img width="1004" alt="image" src="https://github.com/actions/runner-images/assets/103591278/8d38a210-82f2-4e17-8d4c-1f9a5703f524">

**Reply by ilia-shipitsin** [reply]
pardon, what is "deployed image" ?

**Comment by iZnoGod722** [comment]
According to the policy

"Tools and versions will typically be removed 6 months after they are deprecated or have reached end-of-life"

Net Core 3.1 should be removed from the images around June/July period?

**Reply by markatky46** [reply]
@iZnoGod722 thank you for the concise answer. I will feed this back internally.

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
